[Question] Pushing To A Private Registry: Am I Missing Something?

Hi all,

I'm trying to push to a private registry but not having much luck.  The registry is running unsecured for now (while I get going).  My `pom.xml` contains the following (mainly cobbled together from the docs)

    <plugin>
      <groupId>com.spotify</groupId>
      <artifactId>docker-maven-plugin</artifactId>
      <version>0.3.3</version>
      <configuration>
        <imageName>customer-service</imageName>
        <baseImage>java</baseImage>
        <entryPoint>["java", "-jar", "/${project.build.finalName}.jar"]</entryPoint>
        <serverId>office-dockerregistry</serverId>
        <registryUrl>http://office-dockerregistry.openobjects.com:5000/v2/</registryUrl>
        <!-- copy the service's jar file from target into the root directory of the image -->
        <resources>
          <resource>
            <directory>${project.build.directory}</directory>
            <include>${project.build.finalName}.jar</include>
          </resource>
        </resources>
      </configuration>
      <executions>
        <execution>
          <phase>package</phase>
          <goals>
            <goal>build</goal>
          </goals>
        </execution>
      </executions>
    </plugin>

In my `settings.xml` I have

    <server>
      <id>office-dockerregistry</id>
      <username>lee</username>
      <password>password</password>
      <configuration>
        <email>lee@openobjects.com</email>
      </configuration>
    </server>

But when I call `mvn docker:push`, I get an error message containing:

    [ERROR] Failed to execute goal com.spotify:docker-maven-plugin:0.3.3:push (default-cli) on project customer-service-core: Exception caught: Post https://registry-1.docker.io/v2/library/customer-service/blobs/uploads/: token auth attempt for registry: https://auth.docker.io/token?account=lee&scope=repository%3Alibrary%2Fcustomer-service%3Apush%2Cpull&service=registry.docker.io request failed with status: 401 Unauthorized -> [Help 1]

So it looks like the `registryUrl` isn't being picked up. Can anyone see anything obvious I'm missing?

--- a/README.md
+++ b/README.md
@@ -125,7 +125,64 @@ To remove the image named `foobar` run the following command:
 For a complete list of configuration options run:
 `mvn com.spotify:docker-maven-plugin:<version>:help -Ddetail=true`
 
-### Authenticating with Private Registries
+### Using with Private Registries
+
+To push an image to a private registry, Docker requires that the image tag
+being pushed is prefixed with the hostname and port of the registry. For
+example to push `my-image` to `registry.example.com`, the image needs to be
+tagged as `registry.example.com/my-image`.
+
+The simplest way to do this with docker-maven-plugin is to put the registry
+name in the `<imageName>` field, for example
+
+```xml
+<plugin>
+  <groupId>com.spotify</groupId>
+  <artifactId>docker-maven-plugin</artifactId>
+  <configuration>
+    <imageName>registry.example.com/my-image</imageName>
+    ...
+```
+
+Then when pushing the image with either `docker:build -DpushImage` or
+`docker:push`, the docker daemon will push to `registry.example.com`.
+
+Alternatively, if you wish to use a short name in `docker:build` you can use
+`docker:tag` to tag the just-built image with the full registry hostname. For
+example:
+
+```xml
+<plugin>
+  <groupId>com.spotify</groupId>
+  <artifactId>docker-maven-plugin</artifactId>
+  <configuration>
+    <imageName>my-image</imageName>
+    ...
+  </configuration>
+  <executions>
+    <execution>
+      <id>build-image</id>
+      <phase>package</phase>
+      <goals>
+        <goal>build</goal>
+      </goals>
+    </execution>
+    <execution>
+      <id>tag-image</id>
+      <phase>package</phase>
+      <goals>
+        <goal>tag</goal>
+      </goals>
+      <configuration>
+        <image>my-image</image>
+        <newName>registry.example.com/my-image</newName>
+      </configuration>
+    </execution>
+  </executions>
+</plugin>
+```
+
+#### Authenticating with Private Registries
 
 To push to a private Docker image registry that requires authentication, you can put your
 credentials in your Maven's global `settings.xml` file as part of the `<servers></servers>` block.

